Add butcher shop roofs

curstwist opened the pull request:
#### Summary
```SUMMARY: Content "Adds roofs to butcher shops."```

#### Purpose of change
Continues roof project.

#### Additional context
s_butcher, s_butcher_roof & s_butcher_upper_roof.   Includes small modifications to back areas.

This one got the small 2nd story with smoking setup.
![s_butcher](https://user-images.githubusercontent.com/39442864/56840936-92c05600-6858-11e9-93c1-1bfa43dbabb5.PNG)
![s_butcher_roof](https://user-images.githubusercontent.com/39442864/56840938-9522b000-6858-11e9-900c-2a34e6ed4064.PNG)
![s_butcher_upper_roof](https://user-images.githubusercontent.com/39442864/56840940-96ec7380-6858-11e9-8e81-cab3874f8bc7.PNG)

s_butcher_1, s_butcher_roof_1 & s_butcher_upper_roof_1.  Includes small modifications to back areas.
![s_butcher_1](https://user-images.githubusercontent.com/39442864/56840964-bd121380-6858-11e9-8d4e-64a0d0c1a481.PNG)
![s_butcher_roof_1](https://user-images.githubusercontent.com/39442864/56840965-bd121380-6858-11e9-9d57-640f827d5bd9.PNG)
![s_butcher_upper_roof_1](https://user-images.githubusercontent.com/39442864/56840966-bd121380-6858-11e9-9be4-4bf814dc301c.PNG)

s_butcher_2, s_butcher_roof_2.  Basic roof.

![s_butcher_2](https://user-images.githubusercontent.com/39442864/56840987-e59a0d80-6858-11e9-9edd-a61b8d5044b0.PNG)
![s_butcher_roof_2](https://user-images.githubusercontent.com/39442864/56840991-e763d100-6858-11e9-8468-1304b9a39a78.PNG)

Updated s_butcher_2 ground floor per feedback:
adjusted how some spawns were defined & fixed item_groups.
![butcher_2 updated](https://user-images.githubusercontent.com/39442864/56849546-c1274b00-68c3-11e9-88e9-4946d156fa09.PNG)

Night-Pryanik commented:
While I'm not strictly against it, but is there a need for a table with chairs in `s_butcher_2`? There isn't even an oven or allocated cooking place to cook food. Visitors ain't going to eat raw meat, are they?

curstwist commented:
> 
> 
> While I'm not strictly against it, but is there a need for a table with chairs in `s_butcher_2`? There isn't even an oven or allocated cooking place to cook food. Visitors ain't going to eat raw meat, are they?

Yeah that was there when I got here. I can put up some extra shelves or more fridges which makes sense.  Probably should get a stove also.

Xyzarach commented on data/json/itemgroups/food_service.json:
```diff
@@ -976,5 +976,49 @@
+      [ "lunchmeat", 6 ],
+      [ "offal", 10 ],
+      [ "hotdogs_frozen", 6 ],
+      { "item": "lung", "prob": 1, "charges": 2 },
```
```suggestion
      { "item": "lung", "prob": 2, "charges": 5 },
```
You might also want to bump the probability and amount of the new offal types, "prob" so that offal items show up about as much as before and "charges" because you need 5 of one to equal a meat chunk in recipes that take both.

Xyzarach commented on data/json/itemgroups/food_service.json:
```diff
@@ -976,5 +976,49 @@
+      [ "offal", 10 ],
+      [ "hotdogs_frozen", 6 ],
+      { "item": "lung", "prob": 1, "charges": 2 },
+      { "item": "kidney", "prob": 1, "charges": 2 },
```
```suggestion
      { "item": "kidney", "prob": 2, "charges": 5 },
```

Xyzarach commented on data/json/itemgroups/food_service.json:
```diff
@@ -976,5 +976,49 @@
+      [ "hotdogs_frozen", 6 ],
+      { "item": "lung", "prob": 1, "charges": 2 },
+      { "item": "kidney", "prob": 1, "charges": 2 },
+      { "item": "liver", "prob": 1, "charges": 1 },
```
```suggestion
      { "item": "liver", "prob": 4, "charges": 5 },
```
Liver gets higher probability than the other offal items because it has a much noticeably higher probability in the harvest lists.

Xyzarach commented on data/json/itemgroups/food_service.json:
```diff
@@ -976,5 +976,49 @@
+      { "item": "lung", "prob": 1, "charges": 2 },
+      { "item": "kidney", "prob": 1, "charges": 2 },
+      { "item": "liver", "prob": 1, "charges": 1 },
+      { "item": "brain", "prob": 1, "charges": 1 },
```
```suggestion
      { "item": "brain", "prob": 2, "charges": 5 },
```

Xyzarach commented on data/json/itemgroups/food_service.json:
```diff
@@ -976,5 +976,49 @@
+      { "item": "kidney", "prob": 1, "charges": 2 },
+      { "item": "liver", "prob": 1, "charges": 1 },
+      { "item": "brain", "prob": 1, "charges": 1 },
+      { "item": "sweetbread", "prob": 2, "charges": 1 }
```
```suggestion
      { "item": "sweetbread", "prob": 2, "charges": 2 }
```
Sweetbread gets fewer charges for balance reasons. Calories on it are enough that you can make good meals out of the sweetbread instead of using it in something else.

Xyzarach commented on data/json/itemgroups/food_service.json:
```diff
@@ -976,5 +976,49 @@
+    "type": "item_group",
+    "id": "butcher_raw_meat",
+    "items": [
+      [ "meat", 8 ],
```
```suggestion
      [ "stomach_larget", 2 ],
      [ "stomach_boiled", 2 ],
      [ "meat", 8 ],
```

Should add raw and boiled large stomachs to account for green and gray tripe.

curstwist commented:
> 
> 
> Offal was split out into a bunch of other items and doesn't show up in harvest lists anymore, so I've cut offal from the drop list, distributed its probability between the items it's been split into, increased the charges of those drops for balance, and added raw and boiled large stomachs to account for the sale of tripe.

Thanks! changed incorporated.

Xyzarach commented on data/json/itemgroups/food_service.json:
```diff
@@ -976,5 +976,50 @@
+    "type": "item_group",
+    "id": "butcher_raw_meat",
+    "items": [
+      [ "stomach_larget", 2 ],
```
```suggestion
      [ "stomach_large", 2 ],
```

Whoops, sorry, looks like a typo crept in.
